src/java.desktop/macosx/classes/sun/java2d/metal/MTLGraphicsConfig.java

```diff
@@ -0,0 +1,401 @@
+import static sun.java2d.pipe.hw.AccelSurface.RT_TEXTURE;
+import static sun.java2d.pipe.hw.ContextCapabilities.*;
+
+import static sun.java2d.metal.MTLContext.MTLContextCaps.CAPS_EXT_BIOP_SHADER;
+
+public final class MTLGraphicsConfig extends CGraphicsConfig
+        implements AccelGraphicsConfig, SurfaceManager.ProxiedGraphicsConfig
+{
+    private static boolean mtlAvailable;
+    private static ImageCapabilities imageCaps = new MTLImageCaps();
+
+    private static final String mtlShadersLib = AccessController.doPrivileged(
+            (PrivilegedAction<String>) () ->
+                    System.getProperty("java.home", "") + File.separator +
+                            "lib" + File.separator + "shaders.metallib");
+
+
+    private BufferCapabilities bufferCaps;
+    private long pConfigInfo;
+    private ContextCapabilities mtlCaps;
+    private final MTLContext context;
+    private final Object disposerReferent = new Object();
+    private final int maxTextureSize;
+
+    private static native boolean isMetalFrameworkAvailable();
+    private static native boolean tryLoadMetalLibrary(int displayID, String shaderLib);
+    private static native long getMTLConfigInfo(int displayID, String mtlShadersLib);
+
+    /**
+     * Returns maximum texture size supported by Metal. Must be
+     * called under MTLRQ lock.
+     */
+    private static native int nativeGetMaxTextureSize();
+
+    static {
+        mtlAvailable = isMetalFrameworkAvailable();
+    }
+
+    private MTLGraphicsConfig(CGraphicsDevice device,
+                              long configInfo, int maxTextureSize,
+                              ContextCapabilities mtlCaps) {
+        super(device);
+
+        this.pConfigInfo = configInfo;
+        this.mtlCaps = mtlCaps;
+        this.maxTextureSize = maxTextureSize;
+        context = new MTLContext(MTLRenderQueue.getInstance());
+        // add a record to the Disposer so that we destroy the native
+        // MTLGraphicsConfigInfo data when this object goes away
+        Disposer.addRecord(disposerReferent,
+                new MTLGCDisposerRecord(pConfigInfo));
+    }
+
+    @Override
+    public Object getProxyKey() {
+        return this;
+    }
+
+    public SurfaceData createManagedSurface(int w, int h, int transparency) {
+        return MTLSurfaceData.createData(this, w, h,
+                getColorModel(transparency),
+                null,
+                MTLSurfaceData.TEXTURE);
+    }
+
+    public static MTLGraphicsConfig getConfig(CGraphicsDevice device,
+                                              int displayID)
+    {
+        if (!mtlAvailable) {
+            return null;
+        }
+
+        if (!tryLoadMetalLibrary(displayID, mtlShadersLib)) {
+            return null;
+        }
+
+        long cfginfo = 0;
+        int textureSize = 0;
+        MTLRenderQueue rq = MTLRenderQueue.getInstance();
+        rq.lock();
+        try {
+            // getMTLConfigInfo() creates and destroys temporary
```

Comment:
Does it really create and destroy temporary surfaces/contexts?

Reply:
No, it doesn't. I have updated JDK-8263363 to include this correction.